### PR TITLE
Let consumers of data providers access context

### PR DESCRIPTION
**Pull request by axelboc**
- `Provider` becomes `DataProvider`
- `ProviderContext` becomes `DataContext`
- `Context` type becomes `DataContextValue`
- The context and its type are now located in `DataProvider.tsx`
- Add hook `useDataContext` and expose in @h5web/app.
- Document API of @h5web/app in README (built-in providers and context value).

### CODE REVIEW

File: packages/app/README.md [hunk 1]
```diff
@@ -129,3 +210,51 @@ import { getFeedbackMailto } from '@h5web/app';
...
+}
+```
+
+The three stores are created with the
+[react-suspense-fetch](https://github.com/dai-shi/react-suspense-fetch) library,
+which relies on
+[React Suspense](https://reactjs.org/docs/react-api.html#reactsuspense). A
+component that uses one of these stores (e.g.
+`entitiesStore.get('/path/to/entity')`) must have a `Suspense` ancestor to
+manage the loading state.
```
Comment (loichuder) [inline packages/app/README.md:260]: Suggestion: you could add a simple example here.

File: packages/app/README.md [hunk 2]
```diff
@@ -129,3 +210,51 @@ import { getFeedbackMailto } from '@h5web/app';
...
+
+  getExportURL?: (
+    dataset: Dataset<ArrayShape, NumericType>,
+    selection: string | undefined,
+    format: ExportFormat
+  ) => string | undefined;
+
+  addProgressListener: (cb: ProgressCallback) => void;
+  removeProgressListener: (cb: ProgressCallback) => void;
+}
```
Comment (loichuder) [inline packages/app/README.md:251]: It is worth mentioning those ?

File: packages/app/src/providers/mock/MockProvider.tsx [hunk 3]
```diff
@@ -1,18 +1,14 @@
...
-import Provider from '../Provider';
+import DataProvider from '../DataProvider';
 import { MockApi } from './mock-api';
 
-interface Props {
-  children: ReactNode;
-}
-
-function MockProvider(props: Props) {
+function MockProvider(props: PropsWithChildren<Record<never, never>>) {
```
Comment (loichuder) [inline packages/app/src/providers/mock/MockProvider.tsx:7]: Don't you think it would be clearer to keep the `Props` interface here ?

**Review by loichuder — APPROVED** [review]
:+1:

File: packages/app/src/providers/mock/MockProvider.tsx (hunk 3 above)
Comment (axelboc) [reply packages/app/src/providers/mock/MockProvider.tsx:7]: I put back an empty `interface Props {}`

File: packages/app/README.md (hunk 2 above)
Comment (axelboc) [reply packages/app/README.md:251]: Yeah no, you're right.

**Comment by axelboc** [comment]
Thanks for the great suggestions 👍

**Pull request merged**